### PR TITLE
ceph-disk: do not create bluestore wal/db partitions by default

### PR DESCRIPTION
**Pull request by dachary**
Fixes: http://tracker.ceph.com/issues/18291

Signed-off-by: Loic Dachary <loic@dachary.org>

### CODE REVIEW

File: src/ceph-disk/ceph_disk/main.py [hunk 1]
```diff
@@ -1813,6 +1813,43 @@ def set_subparser(subparsers):
...
+            If the --bluestore argument is given, a bluestore objectstore
+            will be used instead of the legacy filestore objectstore.
+
+            When an entire device is prepared for bluestore, two
+            partitions are created. The first partition is for metadata,
+            the second partition is for blocks that contain data.
+
+            Unless explicitly specified with --block.db or
+            --block.wal, the bluestore DB and WAL are files within the
+            the metadata directory. For instance:
```
Comment (liewegas) [inline src/ceph-disk/ceph_disk/main.py]: Oops, actually:

Unless explicitly specified with --block.db or --block.wal, the bluestore DB and WAL data is stored on the main block device.

File: src/ceph-disk/ceph_disk/main.py [hunk 2]
```diff
@@ -1813,6 +1813,43 @@ def set_subparser(subparsers):
...
+
+                ceph-disk prepare --bluestore /dev/sdc
+
+            Will create
+
+                /dev/sdc1 for medata (including db & wal files)
```
Comment (liewegas) [inline src/ceph-disk/ceph_disk/main.py]: s/medata/osd metadata/

File: src/ceph-disk/ceph_disk/main.py [hunk 3]
```diff
@@ -1813,6 +1813,43 @@ def set_subparser(subparsers):
...
+                /dev/sdc2 for block (the rest of the disk)
```
Comment (liewegas) [inline src/ceph-disk/ceph_disk/main.py]: for block, db, and wal data (the rest of the disk)

**Comment by dachary** [comment]
@liewegas doc updated as suggested, thanks !

File: src/ceph-disk/ceph_disk/main.py [hunk 4]
```diff
@@ -1813,6 +1813,44 @@ def set_subparser(subparsers):
...
+
+            Unless explicitly specified with --block.db or
+            --block.wal, the bluestore DB and WAL data is stored on
+            the main block device. For instance:
+
+                ceph-disk prepare --bluestore /dev/sdc
+
+            Will create
+
+                /dev/sdc1 for osd medata
```
Comment (liewegas) [inline src/ceph-disk/ceph_disk/main.py]: s/medata/metadata/

File: src/ceph-disk/ceph_disk/main.py [hunk 5]
```diff
@@ -1813,6 +1813,44 @@ def set_subparser(subparsers):
...
+            three and four respectively. For instance:
+
+                ceph-disk prepare --bluestore \\
+                      --block.db /dev/sdc \\
+                      --block.wal /dev/sdc \\
+                      /dev/sdc
+
+            Will create
+
+                /dev/sdc1 for osd medata
```
Comment (liewegas) [inline src/ceph-disk/ceph_disk/main.py]: and here

**Comment by dachary** [comment]
s/medata/metadata/

**Comment by dachary** [comment]
pushed to ceph-ci, will run the following when it is ready
<pre>
teuthology-suite -k distro --verbose --suite ceph-disk --suite-branch master --ceph wip-18291-bluestore --machine-type vps --priority 1000 machine_types/vps.yaml
</pre>

**Comment by liewegas** [comment]
http://pulpito.ceph.com/sage-2016-12-16_20:11:13-ceph-disk-wip-18291-bluestore-distro-basic-vps/

**Comment by liewegas** [comment]
retest this please

**Comment by dachary** [comment]
s/self.args/self.data.args/

**Comment by dachary** [comment]
jenkins test this please (osd-crush)

**Comment by dachary** [comment]
<pre>
teuthology-suite -k distro --newest 50 --verbose --suite ceph-disk --ceph wip-18291-bluestore --machine-type vps --priority 1000
</pre>

* **running** http://pulpito.ceph.com/loic-2016-12-17_11:30:22-ceph-disk-wip-18291-bluestore-distro-basic-vps/

**Comment by dachary** [comment]
http://pulpito.ceph.com/loic-2016-12-18_01:05:47-ceph-disk-wip-18291-bluestore-distro-basic-vps/

**Pull request merged**